Path/IO.hs

```diff
@@ -330,21 +332,46 @@ listDir path = do
       else Right `liftM` parseAbsFile ipath
   return (lefts items, rights items)
 
--- | Similar to 'listDir', but recursively traverses every sub-directory,
--- and collects all files and directories. This can fail with the same
--- exceptions as 'listDir'.
+-- XXX This can be exported as a utility function for use with walkdir for
+-- excluding symlinks during traversal.
+-- | Check if the given directory path is a symbolic link.
```

Comment:
Let's export it with `@since 1.3.0` annotation indeed. Also not to forget: add it to the changelog.